tests/telemetry/test_telemetry.py

```diff
@@ -240,3 +246,119 @@ def test_virtualdb_table_streaming(duthosts, rand_one_dut_hostname, ptfhost, loc
+
+        if not gnmi_client_pid:
+            pytest.fail("Failed to find PID of gNMI client in PTF docker!")
+
+        logger.info("PID of gNMI client in PTF docker is: '{}'".format(gnmi_client_pid))
+
+        terminate_client_cmd = "kill -9 {}".format(gnmi_client_pid)
+        logger.info("Terminating gNMI client with PID: '{}' ...".format(gnmi_client_pid))
+        terminate_cmd_result = ptfhost.shell(terminate_client_cmd)
+        exit_code = terminate_cmd_result["rc"]
+        pytest_assert(exit_code == 0, "Failed to terminate gNMI client with PID '{}'!".format(gnmi_client_pid))
+        logger.info("gNMI client with PID: '{}' was terminated!".format(gnmi_client_pid))
+
+
+@pytest.fixture
+def setup_testing_memory_spike(duthosts, rand_one_dut_hostname, ptfhost):
+    """Before testing, gNMI client will be started. After testng, gNMI client will be terminated
+    and telemetry container will be restarted.
+
+    Args:
+        duthosts: list of DUTs.
+        rand_one_dut_hostname: The fixture returns a randomly selected DuT.
+        pfthost: PTF docker binding to the selected DuT.
+
+    Returns:
+        None.
+    """
+    duthost = duthosts[rand_one_dut_hostname]
+    dut_ip = duthost.mgmt_ip
+    client_thread = ThreadPool(processes=1)
+    client_thread.apply_async(run_gnmi_client, (ptfhost, dut_ip))
+
+    yield
+
+    terminate_gnmi_client(ptfhost)
+    clear_failed_flag_and_restart(duthost, CONTAINER_NAME)
+
+
+def test_mem_spike(duthosts, rand_one_dut_hostname, ptfhost, setup_testing_memory_spike):
+    """Test whether memory usage on gNMI server will increase or not if gNMI client
+    continuously creates TCP connections but did not explicitly close them.
+
+    Args:
+        duthosts: list of DUTs.
+        rand_one_dut_hostname: The fixture returns a randomly selected DuT.
+        pfthost: PTF docker binding to the selected DuT.
+
+    Returns:
+        None.
+    """
+    logger.info("Start testing the memory spike issue on gNMI server side ...")
+
+    duthost = duthosts[rand_one_dut_hostname]
+    skip_201911_and_older(duthost)
+
+    mem_usage_val = 0
+    for index in range(CHECK_MEM_USAGE_COUNTER):
+        get_mem_usage_cmd = "docker stats --no-stream --format \{{\{{.MemUsage\}}\}} {}".format(CONTAINER_NAME)
+        cmd_result = duthost.shell(get_mem_usage_cmd)
+
+        exit_code = cmd_result["rc"]
+        pytest_assert(exit_code == 0, "Failed to get memory usage of '{}'!".format(CONTAINER_NAME))
+
+        mem_info = cmd_result["stdout_lines"]
+        mem_usage = mem_info[0].split()[0]
+        logger.info("Memory usage of '{}' is: '{}'.".format(CONTAINER_NAME, mem_usage))
+
+        try:
+            mem_val = float(mem_usage[:-3])
+        except ValueError as err:
+            pytest.fail("Failed to convert the memory usage of telemetry from string to float type!")
+
+        pytest_assert(mem_val > mem_usage_val, "Memory usage of telemetry did not increase as expected!")
+        mem_usage_val = mem_val
```

Comment:
We need better determination of memory crazy increasing. How about check `monit` memory alert on telemetry?

Reply:
I updated the test function which uses LogAnalyzer to analyze the syslogs and decide whether the telemetry container was restarted by Monit or not due to the memory spike issue.